src/ocean/io/serialize/StringStructSerializer.d

```diff
@@ -845,4 +846,30 @@ unittest
                          "   ulong a : 1000\n",
             "Incorrect string serializer result");
     }
+
+    // test a typedef array
+
+    struct StructWithTypedefArray
+    {
+        AdskilletId[] ids;
+    }
+
+    StructWithTypedefArray sta;
+
+    sta.ids = new AdskilletId[](4);
+
+    for (int idx = 0; idx < 4; idx++) 
+    {
+        sta.ids[idx] = cast(AdskilletId)(64 + idx);
+    }
+
+    buffer.length = 0;
+    enableStomping(buffer);
+    serializer.serialize(buffer, sta);
+
+    t.test!("==")(buffer.length, 81);
+    t.test(buffer == "struct StructWithTypedefArray:\n" ~
```

Comment:
`t.test!("==")`?

Reply:
All the other tests in this module do == on buffer as well.

Reply:
Feel free to add a commit which switch them then. The advantage of having the operator as template argument is that you get a much more useful error message when it fails.